binary_transparency/firmware/api/annotation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+// SignedStatement is a Statement signed by the Claimant.
+type SignedStatement struct {
+	// The serialised Claim in json form.
+	// This is one of MalwareStatement or BuildStatement.
+	// TODO(mhutchinson): Add an enum for Statement type?
+	Statement []byte
+
+	// Signature is the bytestream of the signature over the Statement.
+	Signature []byte
+}
+
+// FirmwareID is a pointer to a firmware version.
+// TODO(mhutchinson): This could be simplified to just LeafHash or extended to have Revision.
+type FirmwareID struct {
+	LogIndex int64
+	LeafHash []byte
+}
+
+// MalwareStatement is an annotation about malware checks in a firmware version.
+type MalwareStatement struct {
+	// FirmwareID is the FirmwareStatement in the log being annotated.
+	FirmwareID FirmwareID
+
+	// Good is a crude signal of goodness.
+	// TODO(mhutchinson): MVP for reasonable fields. Probably a Timestamp.
```

Comment:
Timestamp super important I suspect

Reply:
Definitely. We should discuss this as a team and pop in whatever we think the minimum set is to demonstrate value. I can imagine arguments for allowing custom metadata here (e.g. some kind of version of the malware scanner or policy engine that was used by the malware checker), but perhaps a timestamp is Good Enough.

Reply:
Leave as-is for now. Leave the TODO for now to add more useful metadata as we need it.